crates/sui-data-ingestion-core/src/reducer.rs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{Reducer, Worker, MAX_CHECKPOINTS_IN_PROGRESS};
+use anyhow::Result;
+use futures::StreamExt;
+use std::collections::HashMap;
+use sui_types::messages_checkpoint::CheckpointSequenceNumber;
+use tokio::sync::mpsc;
+use tokio_stream::wrappers::ReceiverStream;
+
+pub(crate) async fn reduce<W: Worker>(
+    task_name: String,
+    mut current_checkpoint_number: CheckpointSequenceNumber,
+    progress_receiver: mpsc::Receiver<(CheckpointSequenceNumber, W::Result)>,
+    executor_progress_sender: mpsc::Sender<(String, CheckpointSequenceNumber)>,
+    reducer: Option<Box<dyn Reducer<W::Result>>>,
+) -> Result<()> {
+    let mut stream =
+        ReceiverStream::new(progress_receiver).ready_chunks(MAX_CHECKPOINTS_IN_PROGRESS);
```

Comment:
The size of the chunk is not very intuitive. It is the upper bound of how many checkpoints could be in the receiver. Could you add a comment?

Reply:
yes, just a way for everything that is ready to be fetched and processed immediately in a single loop iteration. Will add a code comment